app/components/notification-list-row/component.js

```diff
@@ -0,0 +1,43 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+
+const notificationOptions = [
+    {
+        label: 'components.notificationListRow.instant',
+        value: 'instant',
+    },
+    {
+        label: 'components.notificationListRow.daily',
+        value: 'daily',
+    },
+    {
+        label: 'components.notificationListRow.never',
+        value: 'none',
+    },
+];
+
+export default Component.extend({
+    i18n: service(),
+    toast: service(),
+
+    notificationOptions,
+    currentSettingLabel: computed('subscription.frequency', function () {
+        const frequency = this.get('subscription.frequency');
+        return this.get('notificationOptions').findBy('value', frequency).label;
+    }),
+
+    updateFrequency: task(function* (frequency) {
+        const subscription = this.get('subscription');
+        try {
+            subscription.set('frequency', frequency);
+            if (subscription.changedAttributes()) {
+                yield subscription.save();
```

Comment:
What happens if this yield never gets hit? Does the task keep running?

Reply:
I don't think so. `yield` is similar to `await` in Python, which, according to Ember Concurrency docs, "lets you pause execution of your task function until that promise resolves". I do not think `yield` is by any means necessary in a task. Therefore, if `yield` is not hit, the task would proceed without waiting for a promise.